Fix coreg tutorial typos

jhouck opened the pull request:
#### Reference issue
Minor cleanup, no issue

#### What does this implement/fix?
I noticed a few minor typos (e.g., slashes not quoted) and have addressed. Copy-and-paste into python should now work for this one. Also added a cite for the paper that assessed the reliability of the method.

mmagnuski commented:
The `data_path / 'dirname'` parts are not typos, we now use pathlib objects for paths management. Instead of `op.join(...)` you can simply use slash to join with system-specific separator.

cbrnr commented on tutorials/forward/25_automated_coreg.py:
```diff
@@ -7,7 +7,9 @@
-automated MEG-MRI coregistration via scripting.
+automated MEG-MRI coregistration via scripting. Generally the results of
+this approach are consistent with those obtained from manual
+coregistration :footcite:`HouckClaus2020`
```
```suggestion
coregistration :footcite:`HouckClaus2020`.
```

cbrnr commented on tutorials/forward/25_automated_coreg.py:
```diff
@@ -23,17 +23,17 @@
-
-data_path = mne.datasets.sample.data_path()
-subjects_dir = data_path + '/subjects'
+data_path = Path(mne.datasets.sample.data_path())
```
This is already a `Path` object, so no need to import and convert.

jhouck replied:
That was an attempt to make the use of the `Path` object more obvious. Maybe a comment instead of the redundant import/conversion?

jhouck commented:
@mmagnuski I knew that, and still I did this. I've removed the errors that I'd introduced and added an explicit call to `pathlib.Path` along with @cbrnr's catch.

Also, the tutorial mentions `mne.scale_mri` but I don't see an example or tutorial elsewhere that uses it. Is there one that I'm missing? That might be beyond the scope of this PR. The previous guide was in https://www.slideshare.net/mne-python but things have changed a bit since then.

mmagnuski commented:
I don't think we have an example showing the usage of mne.scale_mri. Might be useful to add in the future.

larsoner commented on tutorials/forward/25_automated_coreg.py:
```diff
@@ -26,8 +28,8 @@
 
-
 data_path = mne.datasets.sample.data_path()
+# data_path and all paths built from it are `pathlib.Path`_ objects
```
I don't think this will render because it's in a code comment :(

Plus I think this isn't specific/relevant enough to the automated coreg stuff to include here, maybe somewhere in `datasets` instead if you want?
```suggestion
```

larsoner commented on tutorials/forward/25_automated_coreg.py:
```diff
@@ -107,3 +109,12 @@
 #           matched to a subject's head digitization. When scaling is desired,
 #           a scaled surrogate MRI should be created using
 #           :func:`mne.scale_mri`.
+#
+# .. LINKS
+#
+# .. _`pathlib.Path`: https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module
```
If we did link to this it would be better to do it via intersphinx like
```
:class:`python:pathlib.Path`
```
```suggestion
```

jhouck replied:
Sounds good, the >80 character link was causing flake8 errors anyway.

larsoner commented:
Thanks @jhouck !
